spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -323,6 +312,40 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
 
 		return taskExecution.getExecutionId();
 	}
+	
+	/**
+	 * Method to merge the task deployment properties from current task execution and
+	 * previous manifest.
+	 * @param taskExecutionInformation Current task execution information
+	 * @param previousManifest previous manifest info. *
+	 * @return mergedTaskDeploymentProperties Final Merged task deployment properties map
+	 */
+	private Map<String, String> getMergedProperties(TaskExecutionInformation taskExecutionInformation,
```

Comment:
We need to use `TaskAnalysisReport` here instead of the custom comparison logic. Not sure why you removed that at first place.

Reply:
I had two thoughts. One is to do changes in TaskAnalyzer to accept the DeploymentProperties instead of TaskManifest, so that we can avoid creating AppDeploymentRequest. Another one to write a custom logic. Are  you suggesting to make changes in TaskAnalyzer in a way to accept only the DeploymentProperties  as input ?

I saw a requirement to create AppDeploymentRequest at last, so was trying to avoid where ever it is created before final AppDeploymentRequest object creation

Reply:
> One is to do changes in TaskAnalyzer to accept the DeploymentProperties instead of TaskManifest

Sure, you can add an additional method in `TaskAnalyzer` to handle this. We don't want to duplicate the properties difference logic away from `TaskAnalyzer`.

Reply:
Sure, I will do that. Thanks